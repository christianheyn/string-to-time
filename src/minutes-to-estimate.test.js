const { minutesToEstimate } = require('./minutes-to-estimate');

describe('minutesToEstimate(minutes, hoursPerDay = 8)', () => {
    it('is a function', () => {
        const actual = typeof minutesToEstimate;
        const expected = 'function';
        expect(actual).toEqual(expected);
    });

    const testMinutesCases = [
        1200,
        30,
        60,
        (60 * 8),
        (60 * 8 * 3),
        -(60 * 8 * 3),
        -30,
        1260,
        500.20,
        50000,
    ];

    it('returns empty object without minutes parameter', () => {
        const actual = minutesToEstimate();
        const expected = { d: 0, h: 0, m: 0 };
        expect(actual).toEqual(expected);
    });

    it('returns object', () => {
        testMinutesCases.forEach((m) => {
            const actual = minutesToEstimate(m);
            expect(actual).toMatchSnapshot();
        });
    });
});
