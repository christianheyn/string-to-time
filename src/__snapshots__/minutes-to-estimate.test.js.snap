// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 1`] = `
Object {
  "d": 2,
  "h": 4,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 2`] = `
Object {
  "d": 0,
  "h": 0,
  "hoursPerDay": 8,
  "m": 30,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 3`] = `
Object {
  "d": 0,
  "h": 1,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 4`] = `
Object {
  "d": 1,
  "h": 0,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 5`] = `
Object {
  "d": 3,
  "h": 0,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 6`] = `
Object {
  "d": -3,
  "h": 0,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 7`] = `
Object {
  "d": 0,
  "h": 0,
  "hoursPerDay": 8,
  "m": -30,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 8`] = `
Object {
  "d": 2,
  "h": 5,
  "hoursPerDay": 8,
  "m": 0,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 9`] = `
Object {
  "d": 1,
  "h": 0,
  "hoursPerDay": 8,
  "m": 20,
}
`;

exports[`minutesToEstimate(minutes, hoursPerDay = 8) returns object 10`] = `
Object {
  "d": 104,
  "h": 1,
  "hoursPerDay": 8,
  "m": 20,
}
`;
